lib/python/ctypes/ctypesgencore/parser/preprocessor.py

```diff
@@ -169,8 +169,26 @@ def parse(self, filename):
                               shell=True,
                               universal_newlines=True,
                               stdout=subprocess.PIPE,
-                              stderr=subprocess.PIPE)
-        ppout, pperr = pp.communicate()
+                              stderr=subprocess.PIPE)        
+        try:
+            ppout, pperr = pp.communicate()
+        except ValueError:
+            # Fix for https://trac.osgeo.org/grass/ticket/3883,
+            # handling file(s) encoded with mac_roman
+            if sys.platform == 'darwin':
+                pp = subprocess.Popen(cmd,
+                                      shell=True,
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+                ppout, pperr = pp.communicate()
+
+                data = ppout.decode('utf8', 'replace')
+                data.replace("\r\n", "\n").replace("\r", "\n")
+
+                ppout = data.replace('\ufffd', '•')
```

Comment:
Why to replace the replacement character? Replacement character seems like a good representation of a replaced character. :-)

Reply:
good point

Reply:
Well, to be true, the bullet character was THE character not recognised (and put back again). But, in this particular context it doesn't really matter which character is there.

This is an example of its original use:
```
#define kICHelper                       "\pHelper•"
```